Acknowledge Stage 0 simplicity (#101)
arthurfiorette/proposal-try-operator: README.md

This proposal addresses the ergonomic challenges of managing multiple, often nested, `try/catch` blocks necessary for handling operations that may fail at various points.

The try block needlessly encloses the protected code in a block. This often prevents straightforward `const` assignment patterns and can reduce readability through additional nesting. The `catch (error) {}` branch is usually where control-flow divergence happens, while the successful path is often linear.

The solution is to add a `try <expression>` operator, a syntax similar to `await <expression>`, which catches any error that occurs when executing its expression and returns it as a value to the caller.

> JavaScript has no existing syntax-level equivalent for in-place exception-to-value conversion at expression position without crossing block or function boundaries.

Reviewer: ```suggestion
JavaScript has no existing equivalent for in-place exception-to-value conversion without using executor callback arguments, which force code to cross function boundaries and create closures. 
```

}
```

This approach often improves readability by cleanly separating the happy path from error handling.

> Control flow remains linear, making it easier to follow, while only exceptional paths require explicit branching.

Reviewer: ```suggestion
Control flow remains linear, making it easier to follow, while only exception paths require explicit branching.
```

> The result can be a more structured, maintainable function where failures are handled concisely without unnecessary indentation.

Reviewer: ```suggestion
The result is a more structured, maintainable function where failures are handled concisely without unnecessary indentation.
```

Unnecessary hesitance.

### Never throws

The `try` operator converts any ECMAScript throw produced while evaluating its operand expression into a `Result.error(...)` value:

```js
const [ok, error, result] = try some.thing()
```

For example:

- If `some` is `undefined`.
- If `thing` is not a function.
- If accessing the `thing` property on `some` throws an error.
- Any other exception that can arise on that line of code.

All these thrown values are captured and encapsulated in the returned `Result`.

> As with other JavaScript constructs, host-level fatal termination conditions are out of scope.

Reviewer: sounds like something only AI would think of, but it's fine.

Author: That one was truly me lol beep bop
